src/libraries/System.Memory/tests/Span/IndexOfAnyInRange.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            foreach (int length in Enumerable.Range(1, 40))
+            {
+                for (int offset = 0; offset < length; offset++)
+                {
+                    int target = -42;
+                    T[] array = Enumerable.Repeat(Create(target - 2), length).ToArray();
+                    array[offset] = Create(target);
+
+                    Assert.Equal(offset, IndexOfAnyInRange(array, Create(target), Create(target)));
+                    Assert.Equal(offset, IndexOfAnyInRange(array, Create(target - 1), Create(target)));
+                    Assert.Equal(offset, IndexOfAnyInRange(array, Create(target), Create(target + 1)));
+                    Assert.Equal(offset, IndexOfAnyInRange(array, Create(target - 1), Create(target + 1)));
+
+                    Assert.Equal(offset, LastIndexOfAnyInRange(array, Create(target), Create(target)));
+                    Assert.Equal(offset, LastIndexOfAnyInRange(array, Create(target - 1), Create(target)));
+                    Assert.Equal(offset, LastIndexOfAnyInRange(array, Create(target), Create(target + 1)));
+                    Assert.Equal(offset, LastIndexOfAnyInRange(array, Create(target - 1), Create(target + 1)));
+                }
+            }
+        }
+
+        [Fact]
+        public void MatrixOfLengthsAndOffsets_AnyExcept_FindsValueAtExpectedIndex()
+        {
+            foreach (int length in Enumerable.Range(1, 40))
+            {
+                for (int offset = 0; offset < length; offset++)
+                {
+                    int target = -42;
+                    T[] array = Enumerable.Repeat(Create(target), length).ToArray();
+                    array[offset] = Create(target - 2);
+
+                    Assert.Equal(offset, IndexOfAnyExceptInRange(array, Create(target), Create(target)));
+                    Assert.Equal(offset, IndexOfAnyExceptInRange(array, Create(target - 1), Create(target)));
+                    Assert.Equal(offset, IndexOfAnyExceptInRange(array, Create(target), Create(target + 1)));
+                    Assert.Equal(offset, IndexOfAnyExceptInRange(array, Create(target - 1), Create(target + 1)));
+
+                    Assert.Equal(offset, LastIndexOfAnyExceptInRange(array, Create(target), Create(target)));
+                    Assert.Equal(offset, LastIndexOfAnyExceptInRange(array, Create(target - 1), Create(target)));
+                    Assert.Equal(offset, LastIndexOfAnyExceptInRange(array, Create(target), Create(target + 1)));
+                    Assert.Equal(offset, LastIndexOfAnyExceptInRange(array, Create(target - 1), Create(target + 1)));
+                }
+            }
+        }
+
+        // Wrappers for {Last}IndexOfAny{Except}InRange that invoke both the Span and ReadOnlySpan overloads,
+        // ensuring they both produce the same result, and returning that result.
+        // This avoids needing to code the same call sites twice in all the above tests.
```

Comment:
nit: it works very well for all scenarios except of the test case where we are supposed to verify that an exception is being thrown for ROS